rules/prefer-t-regex.js

```diff
@@ -156,7 +156,7 @@ const create = context => {
 		};
 
 		// Only fix a statically verifiable equality.
-		if (regex && matchee.type === 'Literal') {
+		if (regex && matchee?.type === 'Literal') {
```

Comment:
If `matches` is undefined, it would be clearer to guard against it right after it's defined.

Same applies to pretty much all the changes.